### PR TITLE
(417): update hcl syntax in doc examples

### LINKED ISSUE #417: Cleanup HCL examples in website as per new Terraform syntax
Terraform syntax previously required quoted variables to refer items in such way:
```hcl
some_value = "${resource.name.field}"
```

Terraform now fires a lot of warning message when using such quoted variables as the new syntax requires it to be as such:

```hcl
some_value = resource.name.field
```

Our website documentation still has a lot of examples written in older syntax and this issue is set to track work of cleanup.

### PR DESCRIPTION
**Pull request by vladdoster**
- format hcl code blocks via `terraform fmt`
- update attribute syntax due to addition of [first class
  expressions](https://www.terraform.io/upgrade-guides/0-12.html#first-class-expressions) in Terraform v0.12

closes   #417 

Signed-off-by: Vladislav Doster <mvdoster@gmail.com>

### CODE REVIEW

File: website/docs/d/external_network.html.markdown [hunk 1]
```diff
@@ -20,12 +20,12 @@ data "vcd_external_network" "tf-external-network" {
...
 resource "vcd_dnat" "tf-nat-rule" {
-  org             = "tf-org"
-  vdc             = "tf-vdc"
+  org = "tf-org"
+  vdc = "tf-vdc"
   # References the external network name from the data source
-  network_name    = "${data.vcd_external_network.tf-external-network.name}"
-  network_type    = "ext"
-  edge_gateway    = "tf-gw"
+  network_name = "${data.vcd_external_network.tf-external-network.name}"
```
Comment (dataclouder) [inline website/docs/d/external_network.html.markdown:26]:
```suggestion
  network_name = data.vcd_external_network.tf-external-network.name
```

File: website/docs/d/independent_disk.html.markdown [hunk 2]
```diff
@@ -41,4 +41,4 @@ The following arguments are supported:
 
 ## Attribute reference
 
-All attributes defined in [independent disk](/docs/providers/vcd/r/independent_disk.html#attribute-reference) are supported.
\ No newline at end of file
+All attributes defined in [independent disk](/docs/providers/vcd/r/independent_disk.html#attribute-reference) are supported.
```
Comment (dataclouder) [inline website/docs/d/independent_disk.html.markdown:44]: What was changed in this line? Can't see the diff.

File: website/docs/d/nsxt_tier0_router.html.markdown [hunk 3]
```diff
@@ -40,4 +40,4 @@ API does not allow to filter by other fields.
 
 ## Attribute reference
 
-* `is_assigned` - Boolean value reflecting if Tier-0 router is already consumed by external network.
\ No newline at end of file
+* `is_assigned` - Boolean value reflecting if Tier-0 router is already consumed by external network.
```
Comment (dataclouder) [inline website/docs/d/nsxt_tier0_router.html.markdown:43]: What was changed in this line? Can't see the diff.

File: website/docs/d/right.html.markdown [hunk 4]
```diff
@@ -68,4 +68,4 @@ The following arguments are supported:
 ## More information
 
 See [Roles management](/docs/providers/vcd/guides/roles_management.html) for a broader description of how roles and
-rights work together.
\ No newline at end of file
+rights work together.
```
Comment (dataclouder) [inline website/docs/d/right.html.markdown:71]: What was changed in this line? Can't see the diff.

File: website/docs/d/role.html.markdown [hunk 5]
```diff
@@ -88,4 +88,4 @@ The following arguments are supported:
 ## More information
 
 See [Roles management](/docs/providers/vcd/guides/roles_management.html) for a broader description of how roles and
-rights work together.
\ No newline at end of file
+rights work together.
```
Comment (dataclouder) [inline website/docs/d/role.html.markdown:91]: What was changed in this line? Can't see the diff.

File: website/docs/d/storage_profile.html.markdown [hunk 6]
```diff
@@ -33,4 +33,4 @@ The following arguments are supported:
 
 ## Attribute reference
 
-This data source exports only `id` field.
\ No newline at end of file
+This data source exports only `id` field.
```
Comment (dataclouder) [inline website/docs/d/storage_profile.html.markdown:36]: What was changed in this line? Can't see the diff.

File: website/docs/r/edgegateway_settings.html.markdown [hunk 7]
```diff
@@ -104,4 +104,4 @@ terraform import vcd_edgegateway_settings.tf-egw my-org.my-vdc.63ed92de-4001-450
 [docs-import]:https://www.terraform.io/docs/import/
 
 After importing, if you run `terraform plan` you will see the rest of the values and modify the script accordingly for 
-further operations.
\ No newline at end of file
+further operations.
```
Comment (dataclouder) [inline website/docs/r/edgegateway_settings.html.markdown:107]: What was changed in this line? Can't see the diff.

File: website/docs/r/lb_app_rule.html.markdown [hunk 8]
```diff
@@ -43,11 +43,12 @@ resource "vcd_lb_app_rule" "example-two" {
   org          = "my-org"
   vdc          = "my-org-vdc"
   name         = "script1"
-  script = <<-EOT
+  script       = <<-EOT
     acl vmware_page url_beg / vmware redirect location https://www.vmware.com/ ifvmware_page
     acl other_page2 url_beg / other2 redirect location https://www.other2.com/ ifother_page2
     acl hello payload(0,6) -m bin 48656c6c6f0a
-  EOT
+EOT
```
Comment (dataclouder) [inline website/docs/r/lb_app_rule.html.markdown:50]: What was changed in this line? Can't see the diff.

File: website/docs/r/nsxt_ipsec_vpn_tunnel.html.markdown [hunk 9]
```diff
@@ -165,4 +165,4 @@ $ terraform import vcd_nsxt_ipsec_vpn_tunnel.imported my-org.my-org-vdc.my-nsxt-
 
 The above would import the `my-ipsec-vpn-tunnel-name` IPsec VPN Tunne config settings that are defined
 on NSX-T Edge Gateway `my-nsxt-edge-gateway` which is configured in organization named `my-org` and
-VDC named `my-org-vdc`.
\ No newline at end of file
+VDC named `my-org-vdc`.
```
Comment (dataclouder) [inline website/docs/r/nsxt_ipsec_vpn_tunnel.html.markdown:168]: What was changed in this line? Can't see the diff.

File: website/docs/r/nsxv_firewall_rule.html.markdown [hunk 10]
```diff
@@ -274,4 +274,4 @@ Import successful!
 
 The resources that were imported are shown above. These resources are now in
 your Terraform state and will henceforth be managed by Terraform.
-```
\ No newline at end of file
+```
```
Comment (dataclouder) [inline website/docs/r/nsxv_firewall_rule.html.markdown:277]: What was changed in this line? Can't see the diff.

File: website/docs/r/role.html.markdown [hunk 11]
```diff
@@ -76,4 +76,4 @@ at this stage will show the difference between the minimal configuration file an
 ## More information
 
 See [Roles management](/docs/providers/vcd/guides/roles_management.html) for a broader description of how roles and
-rights work together.
\ No newline at end of file
+rights work together.
```
Comment (dataclouder) [inline website/docs/r/role.html.markdown:79]: What was changed in this line? Can't see the diff.

File: website/docs/r/vapp_firewall_rules.html.markdown [hunk 12]
```diff
@@ -147,4 +147,4 @@ Now to import vApp network firewall rules with ID 0027c6ae-7d59-457e-b33e-a89e97
 
 ```shell
 $ terraform import vcd_vapp_firewall_rules.imported org-name.vdc-name.urn:vcloud:vapp:77755b9c-5ec9-41f7-aceb-4cf158786482.0027c6ae-7d59-457e-b33e-a89e97f0bdc1
-```
\ No newline at end of file
+```
```
Comment (dataclouder) [inline website/docs/r/vapp_firewall_rules.html.markdown:150]: What was changed in this line? Can't see the diff.

File: website/docs/r/vapp_nat_rules.html.markdown [hunk 13]
```diff
@@ -176,4 +176,4 @@ Now to import vApp network NAT rules with ID 0027c6ae-7d59-457e-b33e-a89e97f0bdc
 
 ```shell
 $ terraform import vcd_vapp_nat_rules.imported org-name.vdc-name.urn:vcloud:vapp:77755b9c-5ec9-41f7-aceb-4cf158786482.0027c6ae-7d59-457e-b33e-a89e97f0bdc1
-```
\ No newline at end of file
+```
```
Comment (dataclouder) [inline website/docs/r/vapp_nat_rules.html.markdown:179]: What was changed in this line? Can't see the diff.

File: website/docs/r/vapp_network.html.markdown [hunk 14]
```diff
@@ -92,4 +92,4 @@ terraform import vcd_vapp_network.imported org-name.vdc-name.vapp-name.network-n
 NOTE: the default separator (.) can be changed using Provider.import_separator or variable VCD_IMPORT_SEPARATOR
 
 The above command would import the vApp Network named `network-name` that is defined on vApp `vapp-name` 
-which is configured in organization named `my-org` and VDC named `my-org-vdc`.
\ No newline at end of file
+which is configured in organization named `my-org` and VDC named `my-org-vdc`.
```
Comment (dataclouder) [inline website/docs/r/vapp_network.html.markdown:95]: What was changed in this line? Can't see the diff.

File: website/docs/r/vapp_org_network.html.markdown [hunk 15]
```diff
@@ -55,4 +55,4 @@ terraform import vcd_vapp_org_network.imported org-name.vdc-name.vapp-name.org-n
 NOTE: the default separator (.) can be changed using Provider.import_separator or variable VCD_IMPORT_SEPARATOR
 
 The above command would import the vApp Org Network named `org-network-name` that is defined on vApp 
-`vapp-name` which is configured in organization named `my-org` and VDC named `my-org-vdc`.
\ No newline at end of file
+`vapp-name` which is configured in organization named `my-org` and VDC named `my-org-vdc`.
```
Comment (dataclouder) [inline website/docs/r/vapp_org_network.html.markdown:58]: What was changed in this line? Can't see the diff.

File: website/docs/r/vm.html.markdown [hunk 16]
```diff
@@ -83,4 +83,4 @@ vcd_vm.TestVm: Import prepared!
 vcd_vm.TestVm: Refreshing state... [id=urn:vcloud:vm:26c04f4d-2185-4a33-8ef9-019768d29003]
 
 Import successful!
-```
\ No newline at end of file
+```
```
Comment (dataclouder) [inline website/docs/r/vm.html.markdown:86]: What was changed in this line? Can't see the diff.

File: website/docs/r/vm_internal_disk.html.markdown [hunk 17]
```diff
@@ -110,4 +110,4 @@ Now to import disk with ID 3001 one could supply this command:
 
 ```shell
 $ terraform import vcd_vm_internal_disk.imported org-name.vdc-name.vapp-name.vm-name.3001
-```
\ No newline at end of file
+```
```
Comment (dataclouder) [inline website/docs/r/vm_internal_disk.html.markdown:113]: What was changed in this line? Can't see the diff.

File: website/docs/r/vm_sizing_policy.html.markdown [hunk 18]
```diff
@@ -121,4 +121,4 @@ Now to import VM sizing policy with ID urn:vcloud:vdcComputePolicy:446d623e-1eec
 
 ```shell
 $ terraform import vcd_vm_sizing_policy.imported org-name.urn:vcloud:vdcComputePolicy:446d623e-1eec-4c8c-8a14-2f7e6086546b
-```
\ No newline at end of file
+```
```
Comment (dataclouder) [inline website/docs/r/vm_sizing_policy.html.markdown:124]: What was changed in this line? Can't see the diff.

**Review by dataclouder — CHANGES_REQUESTED** [review]
Apart from a change request, there are several lines (usually the last line in a block of changes) that are shown as changed although I could not see the difference with the original. I don't know if this is a GitHub glitch or an invisible character being added where it shouldn't.
Please investigate

**Comment by vladdoster** [comment]
Thanks for taking the time to look over this large PR, @dataclouder!

File: website/docs/guides/data_source_filters.html.markdown [hunk 19]
```diff
@@ -44,13 +44,16 @@ saved into a catalog. See Example 7 below.
 When a filter contains multiple clauses, you achieve the overall match only if all the clauses match. For example:
 
 ```hcl
+# TF-UPGRADE-TODO: Block type was not recognized, so this block and its contents were not automatically upgraded.
```
Comment (Didainius) [inline website/docs/guides/data_source_filters.html.markdown:47]: This was injected by the tool here and I guess it doesn't make sense as it is just an excerpt, not a complete HCL.

**Review by Didainius — COMMENTED** [review]
Thanks for this improvement. I have this one comment where I think the tool left out unneeded comment.

**Review by lvirbalas — APPROVED** [review]
LGTM! Thank you for your contribution!

File: website/docs/guides/data_source_filters.html.markdown (hunk 19 above)
Comment (Didainius) [reply website/docs/guides/data_source_filters.html.markdown:47]: I will use the opportunity for one more tiny thing. This is a very good user facing cleanup and it deserves a changelog note :)

Could you add a `705-improvements.md` file to `.changes/v3.4.0` and put inside a note. Feel free to grab inspiration from any other notes files there. 

Thank you!

Comment (Didainius) [reply website/docs/guides/data_source_filters.html.markdown:47]: Ping!

Comment (Didainius) [reply website/docs/guides/data_source_filters.html.markdown:47]: Thanks for the contributoin - I am going to merge it and add a changelog note in a separate pr.

**Review by Didainius — APPROVED** [review]

**Pull request merged**